sponsor-dapp/src/components/ContractInteraction.js

```diff
@@ -47,6 +49,36 @@ class ContractInteraction extends Component {
     );
   }
 
+  calcMaxTokensThatCanBeCreated() {
+    const { drizzleHelper } = this;
+    const { contractAddress } = this.props;
+    const { web3 } = this.props.drizzle;
+
+    const fpScalingFactor = BigNumber(web3.utils.toWei("1", "ether"));
+    const newExcessMargin = BigNumber(drizzleHelper.getCache(contractAddress, "calcExcessMargin", []));
+    const newUnderlyingPrice = BigNumber(
+      drizzleHelper.getCache(contractAddress, "getUpdatedUnderlyingPrice", []).underlyingPrice
+    );
+
+    const derivativeStorage = drizzleHelper.getCache(contractAddress, "derivativeStorage", []);
+    const startingUnderlyingPrice = BigNumber(derivativeStorage.referenceTokenState.underlyingPrice);
+    const startingTokenPrice = BigNumber(derivativeStorage.referenceTokenState.tokenPrice);
+    const supportedMove = BigNumber(derivativeStorage.fixedParameters.supportedMove);
+    const leverage = BigNumber(
+      drizzleHelper.getCache(derivativeStorage.externalAddresses.returnCalculator, "leverage", [])
+    );
+
+    const maxTokens = newExcessMargin
+      .times(startingUnderlyingPrice)
+      .times(fpScalingFactor)
+      .div(startingTokenPrice)
+      .div(newUnderlyingPrice)
+      .div(supportedMove)
+      .div(leverage);
```

Comment:
I think we should either:
- take the absolute value of `leverage` and the `newUnderlyingPrice`.
- take the absolute value of the output.

These numbers can technically be negative and I don't think a negative number would ever make sense in this field.

Reply:
Done.